drivers/SmartThings/zigbee-button/src/zigbee-multi-button/robb/init.lua

```diff
@@ -0,0 +1,226 @@
+local device_management = require "st.zigbee.device_management"
+
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+local Basic = zcl_clusters.Basic
+local Level = zcl_clusters.Level
+local OnOff = zcl_clusters.OnOff
+local PowerConfiguration = zcl_clusters.PowerConfiguration
+local capabilities = require "st.capabilities"
+
+local utils = require "st.utils"
+
+--[[
+The ROBB Wireless Remote Control has 4 or 8 buttons. They are arranged in two columns:
+
+All buttons on the left side support 'pressed' (OnOff > ON) and 'up_hold' (Level > MoveStepMode.UP).
+All buttons on the right side support 'pressed' (OnOff > OFF) and 'down_hold' (Level > MoveStepMode.DOWN).
+
+Each button-row represents one endpoint. The 8x remote control has four endpoints, the 4x remote control has two.
+That means each endpoint has two buttons.
+--]]
+
+local ROBB_MFR_STRING = "ROBB smarrt"
+local WIRELESS_REMOTE_FINGERPRINTS = {
+  ["ROB_200-008-0"] = {
+    endpoints = 2,
+    buttons = 4
+  },
+  ["ROB_200-007-0"] = {
+    endpoints = 4,
+    buttons = 8
+  }
+}
+
+local function can_handle(opts, driver, device, ...)
+  if device:get_manufacturer() == ROBB_MFR_STRING and WIRELESS_REMOTE_FINGERPRINTS[device:get_model()] then
+    return true
+  else
+    return false
+  end
+end
+
+local build_button_handler = function(addF)
+  return function(driver, device, zb_rx)
+    local comp
+    local additional_fields = { state_change = true }
+    local ep = zb_rx.address_header.src_endpoint.value
+
+    -- Fetch correct button name
+    if addF == true then
+      comp = device:get_component_id_for_endpoint(ep + 240)
+    else
+      comp = device:get_component_id_for_endpoint(ep)
+    end
+
+    local event = capabilities.button.button.pushed(additional_fields)
+    local comp = device.profile.components[comp.id]
+    -- Emit events
+    if comp ~= nil then
+      device:emit_component_event(comp, event)
+      if comp.id ~= "main" then
+        device:emit_event(event)
+      end
+    end
+  end
+end
+
+local function hold_handler(driver, device, zb_rx)
+  local comp
+  local pressed_type
+  local additional_fields = { state_change = true }
+  local ep = zb_rx.address_header.src_endpoint.value
+
+  -- Handle MoveStepMode.UP
+  if zcl_clusters.Level.types.MoveStepMode.UP == zb_rx.body.zcl_body.move_mode.value then
+    comp = device:get_component_id_for_endpoint(ep)
+    pressed_type = "up_hold"
+  -- Handle MoveStepMode.DOWN
+  else
+    comp = device:get_component_id_for_endpoint(ep + 240)
```

Comment:
use the same `offset` and `emit_event_for_endpoint` pattern as above

Reply:
Changed with [8fb92ba](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/604/commits/8fb92ba337ac7bcc1e11c99e0dd85f5e5295cac0)